imap/http_jwt.c

```diff
@@ -503,32 +504,60 @@ static int validate_payload(struct jwt *jwt, char *out, size_t outlen)
         goto done;
     }
 
-    if (json_object_size(jws) == 2) {
-        json_t *jiat = json_object_get(jws, "iat");
-        if (!json_is_integer(jiat)) {
-            if (jiat) {
-                char *val = json_dumps(jiat, JSON_COMPACT|JSON_ENCODE_ANY);
-                xsyslog(LOG_ERR, "Invalid \"iat\" claim", "iat=<%s>", val);
-                free(val);
-            }
-            else xsyslog(LOG_ERR, "JWT contains unsupported claims", NULL);
+    // Parse claims
+    int has_err = 0;
+
+    json_t *jnbf = json_object_get(jws, "nbf");
+    if (jnbf && !json_is_integer(jnbf)) {
+        xsyslog(LOG_ERR, "Invalid \"nbf\" claim", NULL);
+        has_err = 1;
+    }
+
+    json_t *jexp = json_object_get(jws, "exp");
+    if (jexp && !json_is_integer(jexp)) {
+        xsyslog(LOG_ERR, "Invalid \"exp\" claim", NULL);
+        has_err = 1;
+    }
+
+    json_t *jiat = json_object_get(jws, "iat");
+    if (jiat && !json_is_integer(jiat)) {
+        xsyslog(LOG_ERR, "Invalid \"iat\" claim", NULL);
+        has_err = 1;
+    }
+
+    if (has_err)
+        goto done;
+
+    // Validate claims
+
+    if (jnbf) {
+        time_t nbf = json_integer_value(jnbf);
+        if (!(now >= nbf)) {
+            xsyslog(LOG_ERR, "Out-of-range \"nbf\" claim", NULL);
```

Comment:
Nit: Why not (now < nbf) ?

Reply:
@bamthomas says [here](https://github.com/cyrusimap/cyrus-imapd/pull/4515#issuecomment-1573429990):

> I used negative checks !(now < exp) and not now >= exp to be [similar to the spec wording](https://www.rfc-editor.org/rfc/rfc7519#section-4.1.5)

I guess I would have done this differently, but I'm fine with the current conditionals, too.